transport/grpc/outbound_test.go

```diff
@@ -200,3 +202,84 @@ func TestCallStreamWithInvalidPeer(t *testing.T) {
 		}.Error(),
 	)
 }
+
+func TestCallSvcMatch(t *testing.T) {
+	tests := []struct {
+		msg         string
+		serverPort  string
+		headerKey   string
+		headerValue string
+		wantErr     bool
+	}{
+		{
+			msg:         "call service match success",
+			serverPort:  "15213",
+			headerKey:   ServiceHeader,
+			headerValue: "Service",
+		},
+		{
+			msg:         "call service match failed",
+			serverPort:  "15619",
+			headerKey:   ServiceHeader,
+			headerValue: "ThisIsWrongSvcName",
+			wantErr:     true,
+		},
+		{
+			msg:        "no service name response header",
+			serverPort: "15640",
```

Comment:
Don't hard code the port. This can cause conflicts/flaky tests.
You can use port `0` to let the OS assign an unused port to you.

```
listener, err := net.Listen("tcp", ":0")
require.NoError(t, err)
```

You get the address the listener is listening on with `listener.Addr()` --
you're already doing this in the code below so just dropping the `serverPort`
field and changing the `net.Listen` to what I said above should work.